extractor/filesystem/language/haskell/cabal/cabal.go

```diff
@@ -171,6 +170,15 @@ func (e Extractor) extractFromInput(ctx context.Context, input *filesystem.ScanI
 
 			pkgs = append(pkgs, i)
 		}
+
+		if s.Err() != nil {
+			return pkgs, fmt.Errorf("error while scanning cabal.project.freeze file from %v: %w", input.Path, s.Err())
+		}
+	}
+
+	// EOF
+	if len(pkgs) == 0 {
+		return pkgs, fmt.Errorf("EOF reached while scanning %q", input.Path)
```

Comment:
So we return an error if no packages were found? Isn't that possible? e.g. if the project has no dependencies.
I'd say we shouldn't return an error in those cases.